brian2/stateupdaters/GSL.py

```diff
@@ -0,0 +1,204 @@
+'''
+Module containg the StateUpdateMethod for integration using the ODE solver
+provided in the GNU Scientific Library (GSL)
+'''
+import sys
+
+from .base import (StateUpdateMethod, UnsupportedEquationsException, extract_method_options)
+from ..core.preferences import prefs
+from ..devices.device import auto_target
+from brian2.utils.logger import get_logger
+
+logger = get_logger(__name__)
+
+__all__ = ['gsl_rk2', 'gsl_rk4', 'gsl_rkf45', 'gsl_rkck', 'gsl_rk8pd']
+
+default_method_options = {
+    'adaptable_timestep': True,
+    'absolute_error': 1e-6,
```

Comment:
As you saw (later comment), we have `absolute_error_per_variable` which uses the units of the respective variable (we cannot use units for `absolute_error`, since it applies to all variables which can have different units). It's a bit unfortunate that our default error is just a fixed number without any particular meaning, at some point I hesitated to force the user to explicitly chose an error. On the other hand, we also chose a "reasonable value" for `defaultclock.dt`, and this is somewhat comparable. E.g. for a membrane potential, `1e-6` corresponds to `1e-3mV` which seems reasonable and happens to be the error that NEST uses (hard-coded in the model) when integrating with GSL.